canvas.py

```diff
@@ -0,0 +1,94 @@
+from random import random
+from kivy.app import App
+from kivy.uix.widget import Widget
+from kivy.uix.button import Button
+from kivy.graphics import Color, Ellipse, Line
+from kivy.uix.floatlayout import FloatLayout
+from functools import partial
+global X, Y
+#Initial coords.
+X = 400
+Y = 400
+
+class Canvas(Widget):
+
+    def paint(self, token, event):
+        color = (1, 1, 1)  #Red Color
+        global X, Y
+
+        #Increase/Decrease of coordinates dependant on token value passed on button press
+
+        if(token == 0):
+            X -= 2
+        if(token == 1):
+            X += 2
+        if(token == 2):
+            Y += 2
+        if(token == 3):
+            Y -= 2
+
+        with self.canvas:
+          
+          #Temporary Boundaries Example
+            if(X>500):
+                X = 600
+            if(Y>500):
+                Y = 600
+
+            Color(*color, mode='hsv')
+            d = 5.  #diameter of ellipse
```

Comment:
Did you mean 5 or 5.0?

Reply:
Its 5, left the dot because of sketchy documentation on kivy. It does not matter, so will delete.